docs/api/json.md

```diff
@@ -0,0 +1,320 @@
+   encoding of a stream of data change events, e.g.:
+   ```json
+   {"delete": {"part": 1, "vendor": 2, "price": 10000}}
+   {"insert": {"part": 1, "vendor": 2, "price": 30000}}
+   {"insert": {"part": 2, "vendor": 3, "price": 34000}}
+   ```
+4. [Configuring JSON event streams](#configuring-json-event-streams).
+   Describes how the user can specify the JSON format for a stream of events
+   when sending and receiving data via connectors or over HTTP.
+
+## Encoding individual rows
+
+A row in a SQL table or view is encoded as a JSON object whose keys match
+the names of the columns in the table.  Keys can occur in an arbitrary order.
+Column names are case-insensitive, except for columns whose names are declared
+in double quotes (e.g., `create table foo("col")`).  `NULL`s are encoded
+as JSON `null` values or by simply omitting the columns whose value is `NULL`.
+
+For example, given the following table declaration
+
+```sql
+create table json_test (
+    b BOOLEAN,
+    i INTEGER,
+    d DOUBLE,
+    v VARCHAR(32),
+    cc CHAR(16),
+    t TIME,
+    ts TIMESTAMP,
+    dt DATE,
+    ar BIGINT ARRAY
+);
+```
+
+the JSON encoding of a row would look like this:
+
+```json
+{
+    "B":true,
+    "I":-1625240816,
+    "D":0.7879946935782574,
+    "V":"quod",
+    "CC":"voluptatem",
+    "T":"05:05:24",
+    "TS":"2023-11-21 23:19:09",
+    "DT":"2495-03-07",
+    "AR":[1,2,3,4,5]
+}
+```
+
+## Types
+
+| Type                                    | Example                                         |
+|-----------------------------------------|-------------------------------------------------|
+| BOOLEAN                                 | `true`, `false`                                 |
+| TINYINT,SMALLINT, INTEGER, BIGINT       |  `1`, `-9`                                      |
+| FLOAT, DOUBLE, DECIMAL                  | `-1.40`, `12.53`, `1e20`                        |
+| VARCHAR, CHAR, STRING                   | `abc`                                           |
+| TIME                                    | `12:12:33`, `23:59:29.483`, `23:59:09.483221092`|
+| TIMESTAMP                               | `2024-02-25 12:12:33`                           |
+| DATE                                    | `2024-02-25`                                    |
+| BIGINT ARRAY                            | `[1, 2]`                                        |
+| VARCHAR ARRAY ARRAY                     | `[[ 'abc', '123'], ['c', 'sql']]`               |
+
+### `BOOLEAN`
+
+The accepted values are `true` or `false`.
+
+### Integers (`TINYINT`, `SMALLINT`, `INTEGER`, `BIGINT`)
+
+Must be a valid integer and fit the range of the type (see [SQL
+Types](../sql/types.md)), otherwise an error is returned on ingress.
+
+### Decimals (`DECIMAL` / `NUMERIC`)
+
+Both the scientific notation (e.g., `3e234`) and standard floating point numbers
+(`1.23`) are valid. The value must fit within the specified range or
+precision, otherwise an error is returned.
```

Comment:
These are named 'precision' and 'scale', and 'scale' should probably not trigger an error. Does it?